### PR TITLE
Add Windows implementation.

### PR DESCRIPTION
**Pull request by rectified95**
# Summary

Implemented the module using React Native for Windows (https://microsoft.github.io/react-native-windows/docs/getting-started).

Implemented the following properties:
- disabled, value, 
- onCheckColor, onTintColor, onFillColor, tintColor (these match iOS)

Closes microsoft/react-native-windows#5636

## Test Plan

Run `yarn run windows` under `react-native-checkbox\example`, then launch the example app solution from Visual Studio (`react-native-checkbox\example\windows\CheckboxExampleWindows.sln`.
demo-windows

### What's required for testing (prerequisites)?
https://microsoft.github.io/react-native-windows/docs/rnw-dependencies

### What are the steps to reproduce (after prerequisites)?
Play around with the example app, as shown above.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| Windows     |    ✅     |

## Checklist

- [X] I have tested this on a device and a simulator
- [X] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [X] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: src/windows/CheckboxWindows/CheckboxView.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    private:
+        Microsoft::ReactNative::IReactContext m_reactContext{ nullptr };
+        bool m_updating{ false };
+
+        xaml::Controls::CheckBox::Checked_revoker m_checkBoxCheckedRevoker{};
+        xaml::Controls::CheckBox::Unchecked_revoker m_checkBoxUncheckedRevoker{};
+
+        void RegisterEvents();
+        void OnCheckedHandler(winrt::Windows::Foundation::IInspectable const& /*sender*/, xaml::RoutedEventArgs const& args, bool isChecked);
+        // int64_t m_maxValue, m_minValue;
```
Comment (kaiguo) [inline src/windows/CheckboxWindows/CheckboxView.h:28]: Remove?

**Comment by rectified95** [comment]
@kaiguo Removed the commented code, did you notice anything else I missed?

**Comment by rectified95** [comment]
Hey, @nicholaslee119! My colleague has signed off on this PR - can you take a look as well?

**Comment by nicholaslee119** [comment]
@rectified95 
Thanks for your PR! as it is a huge PR, please give us time to review it

**Pull request merged**